deeppavlov/core/trainers/fit_trainer.py

```diff
@@ -0,0 +1,260 @@
+            if 'fit_on' in component_config:
+                component: Estimator
+
+                targets = component_config['fit_on']
+                if isinstance(targets, str):
+                    targets = [targets]
+
+                if self.batch_size > 0 and callable(getattr(component, 'partial_fit', None)):
+                    writer = None
+
+                    for i, (x, y) in enumerate(iterator.gen_batches(self.batch_size, shuffle=False)):
+                        preprocessed = self._chainer.compute(x, y, targets=targets)
+                        # noinspection PyUnresolvedReferences
+                        result = component.partial_fit(*preprocessed)
+
+                        if result is not None and self.tensorboard_log_dir is not None:
+                            if writer is None:
+                                writer = self._tf.summary.FileWriter(str(self.tensorboard_log_dir /
+                                                                         f'partial_fit_{component_index}_log'))
+                            for name, score in result.items():
+                                summary = self._tf.Summary()
+                                summary.value.add(tag='partial_fit/' + name, simple_value=score)
+                                writer.add_summary(summary, i)
+                            writer.flush()
+                else:
+                    preprocessed = self._chainer.compute(*iterator.get_instances(), targets=targets)
+                    if len(targets) == 1:
+                        preprocessed = [preprocessed]
+                    result: Optional[Dict[str, Iterable[float]]] = component.fit(*preprocessed)
+
+                    if result is not None and self.tensorboard_log_dir is not None:
+                        writer = self._tf.summary.FileWriter(str(self.tensorboard_log_dir /
+                                                                 f'fit_log_{component_index}'))
+                        for name, scores in result.items():
+                            for i, score in enumerate(scores):
+                                summary = self._tf.Summary()
+                                summary.value.add(tag='fit/' + name, simple_value=score)
+                                writer.add_summary(summary, i)
+                        writer.flush()
+
+                component.save()
+
+            if 'in' in component_config:
+                c_in = component_config['in']
+                c_out = component_config['out']
+                in_y = component_config.get('in_y', None)
+                main = component_config.get('main', False)
+                self._chainer.append(component, c_in, c_out, in_y, main)
+        self._built = True
+
+    def _load(self) -> None:
+        if not self._loaded:
+            self._chainer.destroy()
+            self._chainer = build_model({'chainer': self.chainer_config}, load_trained=self._saved)
+            self._loaded = True
```

Comment:
Should the `self._loaded` be `True` if `self._saved` is `False`?

Reply:
For when we use the trainer only to evaluate a pipeline without training it.